### PR TITLE
feat: implement NotificationCacheManager for cross-feature cache coordination (FAB-269)

### PR DESCRIPTION
**Pull request by vedovelli-ai-team-sr**
Implements Linear FAB-269

## Summary

NotificationCacheManager is a utility that coordinates TanStack Query cache invalidation across the notifications, tasks, and preferences domains, preventing stale data when notification state changes ripple through the rest of the app.

## What's Included

1. **NotificationCacheManager Class** ()
   - Cross-feature cache coordination utility
   - Methods:
     - `invalidateOnPreferenceChange()` - Invalidate notifications when preferences update
     - `invalidateOnTaskUpdate(taskId)` - Invalidate task detail + notifications queries
     - `filterDisabledNotifications()` - Pure function to filter disabled notification types
     - `syncUnreadCount()` - Recompute unread count from cache without refetch

2. **useNotificationCacheManager Hook** ()
   - React hook providing bound manager methods
   - Automatic queryClient binding via useQueryClient()
   - Exported from src/hooks/index.ts

3. **Unit Tests** ()
   - Comprehensive test coverage for all public methods
   - Edge cases: empty cache, all disabled, unknown types
   - Tests for pagination and multi-page unread count aggregation

4. **Documentation** ()
   - Complete API reference with usage examples
   - Architecture overview and patterns
   - Integration examples for common use cases
   - Performance considerations and migration guide

## Implementation Notes

- Follows CacheInvalidationManager pattern from FAB-60
- Coordinates these query keys:
  - `['notifications']` (FAB-179)
  - `['notificationPreferences']` (FAB-188)
  - `['tasks', taskId]`
  - `['sprints', sprintId, 'metrics']` (FAB-95)
- Pure functions with no side effects
- ~150 lines for manager + hook (matches spec)
- All acceptance criteria met

## Commits

1. `11201d6` - feat: implement NotificationCacheManager utility class
2. `13b71e0` - feat: create useNotificationCacheManager hook
3. `7168fc2` - test: add comprehensive unit tests
4. `367ed1c` - docs: add implementation guide

### CODE REVIEW

File: src/utils/notification-cache-manager.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    return notifications.filter((notification) => {
+      const notificationType = notification.type as keyof NotificationPreferences
+      const typePreference = preferences[notificationType]
+
+      // If preference exists and is disabled, filter out the notification
+      if (typePreference && typeof typePreference === 'object' && 'enabled' in typePreference) {
+        return (typePreference as any).enabled !== false
+      }
+
+      // Default: keep notification if no preference found
```
Comment (vedovelli-ai-team-jr) [inline src/utils/notification-cache-manager.ts]: Love that unknown types stay—safe default when new notification types get added. Prevents accidentally silencing them. Smart.
